operations/helm/charts/mimir-distributed/templates/alertmanager/alertmanager-statefulset.yaml

```
@@ -35,13 +35,21 @@ spec:
           {{ toYaml .Values.alertmanager.persistentVolume.annotations | nindent 10 }}
         {{- end }}
       spec:
+        {{- if $rolloutZone.storageClass}}
+        {{- if (eq "-" $rolloutZone.storageClass) }}
+        storageClassName: {{ .Values.alertmanager.persistentVolume.storageClass }}
+        {{- else }}
+        storageClassName: {{ $rolloutZone.storageClass }}
+        {{- end }}
+        {{- else }}
         {{- if .Values.alertmanager.persistentVolume.storageClass }}
         {{- if (eq "-" .Values.alertmanager.persistentVolume.storageClass) }}
         storageClassName: ""
         {{- else }}
         storageClassName: "{{ .Values.alertmanager.persistentVolume.storageClass }}"
         {{- end }}
         {{- end }}
+        {{- end }}
```

related to [this comment](https://github.com/grafana/mimir/pull/4234#discussion_r1138901327). If I set my values like so:

```
alertmanager.storageClass: "my-storage-class"
alertmanager.zoneAwareReplication.zones[0].storageClass: "-"
```

Then the chart will render the first StatefulSet with my-storage-class instead of ""

```suggestion
        {{- $storageClass := default .Values.alertmanager.persistentVolume.storageClass $rolloutZone.storageClass }}
        {{- if $storageClass }}
        {{- if (eq "-" $storageClass) }}
        storageClassName: ""
        {{- else }}
        storageClassName: {{ $storageClass }}
        {{- end }}
        {{- end }}
```

we can also simplify this using the `default` function